backend/src/agents/datastore_agent.py

```diff
@@ -1,41 +1,39 @@
-from .tool import tool_metadata
-from .types import Parameter
+from src.utils import call_model
+from src.utils.graph_db_utils import create_query
 from src.agents import Agent, agent_metadata
+from src.agents.graph_schema import graph_schema, narrative_options
+import logging
+from src.prompts import PromptEngine
+from datetime import datetime
+from src.utils import to_json
 
+logger = logging.getLogger(__name__)
 
-@tool_metadata(
-    name="last month data",
-    description="Get spending for the last month on a merchant",
-    parameters={
-        "merchant_name": Parameter(
-            type="string",
-            description="Merchant name eg. Spotify",
-        )
-    },
-)
-def get_month_data(merchant_name: str) -> str:
-    if merchant_name.__contains__("Amazon"):
-        return "Last month you spend £64.21 on Amazon"
-    if merchant_name.__contains__("Netflix"):
-        return "Last month you spend £6.99 on Netflix"
-    return f"Last month you didn't spend anything on {merchant_name}"
-
-
-data_store_prompt = """
-You are an expert database agent. You can help with database queries and are connected to a neo4j database.
-
-When looking for an input to any tool being used, a single merchant should be provided.
-
-e.g. "Calculate the sum of Netflix spend over the last month"
-Merchant: Netflix
-"""
+current_user = "John Doe"
+engine = PromptEngine()
 
+generate_cypher_query_prompt = engine.load_prompt("generate-cypher-query",
+                                                  graph_schema=graph_schema,
+                                                  narrative_options=narrative_options,
+                                                  current_date=datetime.now())
 
 @agent_metadata(
     name="DatastoreAgent",
     description="This agent is responsible for handling database queries.",
-    prompt=data_store_prompt,
-    tools=[get_month_data],
+    prompt=generate_cypher_query_prompt,
+    tools=[],
```

Comment:
I am wondering if it's right to remove tools from the agent (even if the an agent only has one tool available)?

If we ever need to add a 2nd tool to this agent, we'd need to re-factor the class and prompts to support multiple tools. If we already supported tools, we'd just need to add a 2nd tool to the list.

I appreciate this might be an additional LLM call but at least we're confident it's selecting the correct tool for the task.

Reply:
Good point and I agree, I have reintroduced the tools